modules/transactionpool/accept.go

```diff
@@ -85,51 +87,37 @@ func relatedObjectIDs(ts []types.Transaction) []ObjectID {
 	return oids
 }
 
-// checkMinerFees checks that the total amount of transaction fees in the
-// transaction set is sufficient to earn a spot in the transaction pool.
-func (tp *TransactionPool) checkMinerFees(ts []types.Transaction) error {
-	// Transactions cannot be added after the TransactionPoolSizeLimit has been
-	// hit.
-	if tp.transactionListSize > TransactionPoolSizeLimit {
-		return errFullTransactionPool
-	}
-
-	// The first TransactionPoolSizeForFee transactions do not need fees.
-	if tp.transactionListSize > TransactionPoolSizeForFee {
-		// Currently required fees are set on a per-transaction basis. 2 coins
-		// are required per transaction if the free-fee limit has been reached,
-		// adding a larger fee is not useful.
-		var feeSum types.Currency
-		for i := range ts {
-			for _, fee := range ts[i].MinerFees {
-				feeSum = feeSum.Add(fee)
-			}
-		}
-		feeRequired := TransactionMinFee.Mul64(uint64(len(ts)))
-		if feeSum.Cmp(feeRequired) < 0 {
-			return errLowMinerFees
-		}
-	}
-	return nil
+// requiredFeesToExtendTpool returns the amount of fees required to extend the
+// transaction pool to fit another transaction set. The amount returned has the
+// unit 'currency per byte'.
+func (tp *TransactionPool) requiredFeesToExtendTpool() types.Currency {
+	// If the transaction pool is nearly empty, it can be extended even if there
+	// are no fees.
+	if tp.transactionListSize < TransactionPoolSizeForFee {
+		return types.ZeroCurrency
+	}
+
+	// Calculate the fee required to bump out the size of the transaction pool.
+	numer := TransactionPoolSizeTarget
+	denom := tp.transactionListSize
+	for i := 1; i < TransactionPoolExponentiation; i++ {
+		numer *= TransactionPoolSizeTarget
+		denom *= tp.transactionListSize
+	}
```

Comment:
```go
ratioToTarget := TransactionPoolSizeTarget / float64(tp.transactionListSize)
feeFactor := math.Pow(ratioToTarget, TransactionPoolExponentiation)
```
I'm confused -- doesn't this mean that as `tp.transactionListSize` increases, the required fee *decreases*? Shouldn't it be the other way around?